code/game/gamemodes/cult/runes.dm

```diff
@@ -1,45 +1,133 @@
 
-///Teleport: Teleports the user to a random teleport rune with the same rune. Can be imbued into a talisman.
+TO ADD A NEW RUNE:
 
+1. Define a new subtype where the others are stored, including the invocation procs.
+2. Add an entry to the switch in determine_icon().
 
-/obj/effect/rune/proc/teleport(key)
-	var/mob/living/user = usr
-	var/allrunesloc[]
-	allrunesloc = new/list()
-	var/index = 0
-	//	var/tempnum = 0
-	for(var/obj/effect/rune/R in world)
-		if(R == src)
-			continue
-		if(R.word1 == wordtravel && R.word2 == wordself && R.word3 == key && R.z != ZLEVEL_CENTCOM)
-			index++
-			allrunesloc.len = index
-			allrunesloc[index] = R.loc
-	if(index >= 5)
-		user << "<span class='danger'>You feel pain, as rune disappears in reality shift caused by too much wear of space-time fabric</span>"
-		if (istype(user, /mob/living))
-			user.take_overall_damage(5, 0)
+*/
+
+/obj/effect/rune
+	name = "rune"
+	desc = "A strange collection of symbols drawn in blood."
+	var/cultist_desc = "An malformed, improper rune. Invoking this may cause problems." //This is shown to cultists who examine the rune in order to determine its true purpose.
+	anchored = 1
+	icon = 'icons/obj/rune.dmi'
+	icon_state = "1"
+	unacidable = 1
+	layer = TURF_LAYER
+
+	var/invocation = "Aiy ele-mayo!" //This is said by cultists when the rune is invoked.
+	var/type = "invalid" //This is used to determine the icon and color of the rune after it is drawn.
+	var/cultists_required = 1 //The amount of cultists required around the rune to invoke it. If only 1, any cultist can invoke it.
+
+/obj/effect/rune/New()
+	..()
+	determine_icon()
+
+/obj/effect/rune/examine(mob/user)
+	..()
+	if(iscultist(user))
+		user << "[cultist_desc]"
+
+/obj/effect/rune/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/weapon/tome) && iscultist(user))
+		user << "<span class='notice'>You carefully erase the rune.</span>"
 		qdel(src)
-	if(allrunesloc && index != 0)
-		if(istype(src,/obj/effect/rune))
-			user.say("Sas[pick("'","`")]so c'arta forbici!")//Only you can stop auto-muting
-		else
-			user.whisper("Sas[pick("'","`")]so c'arta forbici!")
-		user.visible_message("<span class='danger'>[user] disappears in a flash of red light!</span>", \
-		"<span class='userdanger'>You feel as your body gets dragged through the dimension of Nar-Sie!</span>", \
-		"<span class='italics'>You hear a sickening crunch and sloshing of viscera.</span>")
-		user.loc = allrunesloc[rand(1,index)]
 		return
-	if(istype(src,/obj/effect/rune))
-		return fizzle(user) //Use friggin manuals, Dorf, your list was of zero length.
-	else
-		call(/obj/effect/rune/proc/fizzle)()
+	else if(istype(I, /obj/item/weapon/nullrod))
+		user << "<span class='notice'>You disrupt the vile magic with [I].</span>"
+		qdel(src)
 		return
+	return
+
+/obj/effect/rune/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You aren't able to understand the words of [src].</span>"

```

Comment:
lmao anonus you are suggesting xhuis make his code worse.
if you add the here it will result in the the rune